apps/calendar/js/bridge.js

```diff
@@ -52,16 +52,14 @@ exports.observeCalendars = function() {
   });
 
   calendarStore.on('add', getAllAndWrite);
-  calendarStore.on('preRemove', getAllAndWrite);
   calendarStore.on('remove', getAllAndWrite);
   calendarStore.on('update', getAllAndWrite);
 
   stream.cancel = function() {
     calendarStore.off('add', getAllAndWrite);
-    calendarStore.off('preRemove', getAllAndWrite);
     calendarStore.off('remove', getAllAndWrite);
     calendarStore.off('update', getAllAndWrite);
-    stream._emitter.removeAllListeners();
+    stream._cancel();
```

Comment:
on threads.js we really don't need to call anything on the stream itself.. the `ClientStream` handles all the state/cancellation logic (and makes sure no other methods are executed if the thread was cancelled before).

this is a poor's man solution that will go away as soon as we use the real threads.js lib.. so don't worry about it.